bin/init.sh

```diff
@@ -42,55 +44,81 @@ if [ ${ROOT} != "${GO_TOP:-$HOME/go}/src/istio.io/istio" ]; then
        exit 1
 fi
 
-PROXYVERSION=$(grep envoy-debug pilot/docker/Dockerfile.proxy_debug  |cut -d: -f2)
-PROXY=debug-$PROXYVERSION
-
-# Save envoy in $out
-if [ ! -f $OUT/envoy-$PROXYVERSION ] ; then
-    mkdir -p $OUT
-    pushd $OUT
-    # New version of envoy downloaded. Save it to cache, and clean any old version.
+# Download Envoy debug and release binaries for Linux x86_64. They will be included in the
+# docker images created by Dockerfile.proxy and Dockerfile.proxy_debug.
 
-    DOWNLOAD_COMMAND=""
+# Gets the download command supported by the system (currently either curl or wget)
+DOWNLOAD_COMMAND=""
+set_download_command () {
+    # Try curl.
     if command -v curl > /dev/null; then
-       if curl --version | grep Protocols  | grep https > /dev/null; then
-	   DOWNLOAD_COMMAND='curl -fLSs'
-       else
-           echo curl does not support https, will try wget for downloading files.
-       fi
+        if curl --version | grep Protocols  | grep https > /dev/null; then
+	       DOWNLOAD_COMMAND='curl -fLSs'
+	       return
+        fi
+        echo curl does not support https, will try wget for downloading files.
     else
-       echo curl is not installed, will try wget for downloading files.
+        echo curl is not installed, will try wget for downloading files.
     fi
 
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        if command -v wget > /dev/null; then
-	    DOWNLOAD_COMMAND='wget -qO -'
-        else
-            echo wget is not installed.
-        fi
+    # Try wget.
+    if command -v wget > /dev/null; then
+        DOWNLOAD_COMMAND='wget -qO -'
+        return
     fi
+    echo wget is not installed.
 
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        echo Error: curl is not installed or does not support https, wget is not installed. \
-             Cannot download envoy. Please install wget or add support of https to curl.
-        exit 1
-    fi
+    echo Error: curl is not installed or does not support https, wget is not installed. \
+         Cannot download envoy. Please install wget or add support of https to curl.
+    exit 1
+}
+
+# Keep the default URLs up-to-date with the latest push from istio/proxy.
+export ISTIO_ENVOY_DIR=${GO_TOP}/out
```

Comment:
ISTIO_OUT might be more appropriate.

I doubt these exports are going to do much good when run via make.  The only way I could see these carry over to the other scripts that consume them is if someone runs bin/init.sh directly followed by other commands.

Regardless, istio.deps probably needs to be updated to point to this file on what to update.